crates/notion-core/src/serial/package.rs

```diff
@@ -0,0 +1,117 @@
+extern crate serde;
+extern crate serde_json;
+
+use package_info::PackageInfo;
+
+use serde::de::{Deserialize, Deserializer, Error, MapAccess, Visitor};
+
+use std::collections::HashMap;
+use std::fmt;
+use std::hash::Hash;
+use std::marker::PhantomData;
+use std::ops::{Deref, DerefMut};
+
+// wrapper for HashMap to use with deserialization
+pub struct BinMap<K, V>(HashMap<K, V>);
+
+impl<K, V> Deref for BinMap<K, V> {
+    type Target = HashMap<K, V>;
+
+    fn deref(&self) -> &HashMap<K, V> {
+        &self.0
+    }
+}
+
+impl<K, V> DerefMut for BinMap<K, V> {
+    fn deref_mut(&mut self) -> &mut HashMap<K, V> {
+        &mut self.0
+    }
+}
+
+#[derive(Deserialize)]
+pub struct Info {
+    pub name: String,
+    // the "bin" field can be a map or a string
+    // (see https://docs.npmjs.com/files/package.json#bin)
+    #[serde(default)] // handles Option
+    pub bin: Option<BinMap<String, String>>,
+}
+
+impl Info {
+    pub fn into_package_info(self) -> PackageInfo {
+        let mut map = HashMap::new();
+        if self.bin.is_some() {
```

Comment:
Good call, I cleaned up the syntax for that.

And I still have to use `.iter()`, otherwise I get the error:

```
&serial::package::BinMap<std::string::String, std::string::String>` is not an iterator; maybe try calling `.iter()` or a similar method
```